src/include/lru.h

```diff
@@ -174,7 +174,6 @@ class LRU {
 
       // move to pintail
       pintail.push_front(&p->lru_link);
-      adjust();
```

Comment:
This loop probably isn't well tested because `bottom` will be empty if we are in this loop. I think you're right it may be possible for `adjust()` to crash if we remove an item from `top` here. Is that the problem you found originally? If that's the case, then we should fix `adjust()` to handle `bottom.size() == 0`, not remove this call here.

Reply:
int adjust():  
  uint64_t toplen = top.size();
  uint64_t topwant = (midpoint * (double)(lru_get_size() - num_pinned));

both lru_get_size() and num_pinned do not change when moving item to pintail

Reply:
Great point. Thanks Zheng.